boards/shields/rk055hdmipi4m/rk055hdmipi4m.overlay

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2023, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/dt-bindings/display/panel.h>
+
+/{
+	aliases {
+		kscan0 = &touch_controller;
+	};
+
+	chosen {
+		zephyr,display = &lcdif;
+		zephyr,keyboard-scan = &touch_controller;
+	};
+
+	en_mipi_display: enable-mipi-display {
+		compatible = "regulator-fixed";
+		regulator-name = "en_mipi_display";
+		enable-gpios = <&mipi_connector 32 GPIO_ACTIVE_HIGH>;
+		regulator-boot-on;
+	};
+};
+
+&mipi_i2c {
+	status = "okay";
+	touch_controller: gt911@5d {
+		compatible = "goodix,gt911";
+		reg = <0x5d>;
+		irq-gpios = <&mipi_connector 29 GPIO_ACTIVE_HIGH>;
+		reset-gpios = <&mipi_connector 28 GPIO_ACTIVE_HIGH>;
+	};
+};
+
+&lcdif {
+	status = "okay";
+	width = <720>;
+	height = <1280>;
+	display-timings {
+		compatible = "zephyr,panel-timing";
+		hsync-len = <8>;
+		hfront-porch = <32>;
+		hback-porch = <32>;
+		vsync-len = <2>;
+		vfront-porch = <16>;
+		vback-porch = <14>;
+		hsync-active = <0>;
+		vsync-active = <0>;
+		de-active = <1>;
+		pixelclk-active = <1>;
+		/*
+		 * Pixel clock is given by the following formula:
+		 * (height + vsync-len + vfront-porch + vback-porch) *
+		 * (width + hsync-len + hfront-porch + hback-porch) * frame rate
+		 */
+		clock-frequency = <62346240>;
+	};
+	pixel-format = <PANEL_PIXEL_FORMAT_BGR_565>;
+	data-bus-width = "24-bit";
+	backlight-gpios = <&mipi_connector 0 GPIO_ACTIVE_HIGH>;
+};
+
+&mipi_dsi {
```

Comment:
https://docs.zephyrproject.org/latest/hardware/porting/shields.html#board-compatibility, do we consider mipi_dsi to be well-known connector? Should it not be zephyr_mipi_dsi?

Reply:
> The alternative here would be to provide no shield overlay, and only board specific overlays within the shield, since the connection between these displays and the EVK does not use the Arduino headers, and is made via a 40 pin flat flexible cable. I think that defining this cable's pins as a GPIO nexus makes sense though, as the interface is shared between multiple NXP EVKs, and defining the shield this way allows more reuse than board specific overlays

I know that it does not use Arduino headers or other well-known connectors. But IMO mipi_dsi is too SoC specific, there could be mipi_dsi0 and mipi_dsiX, therefore I think it make sense to use zephyr_mipi_dsi nodelabel at the board level as reference to default interface, like we do it using zephyr_udc0.

Reply:
Ok, I'm fine using that. We should do the same to the LCD controller, since the bindings for that are also generic with #53907

Do these node labels look ok to you?
`zephyr_lcdif`- for LCD interface controller
`zephyr_mipi_dsi`- for MIPI DSI controller
`mipi_connector`- for 40 pin connector (present on NXP EVKs, but any board could implement this)
`mipi_i2c`- for I2C interface exposed on the mipi connector. Used with the touch controller.

Reply:
> Ok, I'm fine using that. We should do the same to the LCD controller, since the bindings for that are also generic with #53907
> 
> Do these node labels look ok to you? `zephyr_lcdif`- for LCD interface controller `zephyr_mipi_dsi`- for MIPI DSI controller

LGTM, not sure if you need to prefix lcdif if it is referenced by mipi dsi node. @erwango ?

Reply:
> LGTM, not sure if you need to prefix lcdif if it is referenced by mipi dsi node. @erwango ?

TBH, I think I'm missing some background on this topic, so looks fine to me.

Reply:
The display pipeline here works as follows:

LCDIF writes DPI packets to MIPI, which serializes those packets as DSI data and sends them to the RK055HDMIPI4M. This allows the display to be constantly refreshed, which is required as this is a video mode display. I believe this same pipeline is possible on ST devices using the LTDC and the MIPI-DSI. The only property here that is NXP specific is the `nxp,lcdif`. This is used by our MIPI driver to extract LCD configuration parameters, and avoids the display controller having to redeclare these parameters when using the `mipi_dsi_attach` function.